qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -0,0 +1,480 @@
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            draw_options = dict()
+            for depopt in deprecated:
+                if depopt == "style":
+                    for k, v in fields.pop("style").items():
+                        draw_options[k] = v
+                else:
+                    draw_options[depopt] = fields.pop(depopt)
+            self.drawer.set_options(**draw_options)
+
+        super().set_options(**fields)
+
+    @abstractmethod
+    def _generate_fit_guesses(
+        self,
+        user_opt: FitOptions,
+        curve_data: CurveData,
+    ) -> Union[FitOptions, List[FitOptions]]:
+        """Create algorithmic guess with analysis options and curve data.
+
+        Args:
+            user_opt: Fit options filled with user provided guess and bounds.
+            curve_data: Formatted data collection to fit.
+
+        Returns:
+            List of fit options that are passed to the fitter function.
+        """
+
+    def _format_data(
+        self,
+        curve_data: CurveData,
+    ) -> CurveData:
+        """Post-processing for fit data collection.
+
+        Args:
+            curve_data: Raw data collection created from experiment results.
+
+        Returns:
+            Formatted data.
+        """
+        # take average over the same x value by keeping sigma
+        data_allocation, xdata, ydata, sigma, shots = multi_mean_xy_data(
+            series=curve_data.data_allocation,
+            xdata=curve_data.x,
+            ydata=curve_data.y,
+            sigma=curve_data.y_err,
+            shots=curve_data.shots,
+            method="shots_weighted",
+        )
+
+        # sort by x value in ascending order
+        data_allocation, xdata, ydata, sigma, shots = data_sort(
+            series=data_allocation,
+            xdata=xdata,
+            ydata=ydata,
+            sigma=sigma,
+            shots=shots,
+        )
+
+        return CurveData(
+            x=xdata,
+            y=ydata,
+            y_err=sigma,
+            shots=shots,
+            data_allocation=data_allocation,
+            labels=curve_data.labels,
+        )
+
+    def _evaluate_quality(
+        self,
+        fit_data: FitData,
+    ) -> Union[str, None]:
+        """Evaluate quality of the fit result.
+
+        Args:
+            fit_data: Fit outcome.
+
+        Returns:
+            String that represents fit result quality. Usually "good" or "bad".
+        """
+        if fit_data.reduced_chisq < 3.0:
+            return "good"
+        return "bad"
+
+    def _run_data_processing(
+        self,
+        raw_data: List[Dict],
+        series: List[SeriesDef],
+    ) -> CurveData:
+        """Perform data processing from the experiment result payload.
+
+        Args:
+            raw_data: Payload in the experiment data.
+            series: List of series definition defining filtering condition.
+
+        Returns:
+            Un-formatted data collection.
```

Comment:
I'm looking for better naming for two data types.

- data1: Data generated by data processor. Though y values are transformed into desired format (e.g. counts, probability, projected IQ value, etc...) the data ordering is random (conforms to circuit ordering) and there might be multiple elements for the same x value, e.g. RB, QV; if you run circuit with multiple seed, or when you combine two experiment results.
- data2: after `_format_data`. Ready for fitting. Developer can design arbitrary post-processing, for example, filtering, smoothing, removing outlier. By defaults data ordering is corrected and duplicated x values are averaged.

This will be described in module documentation but I think still intuitive naming will help developer.

Reply:
How about:

* Processed Data: The data output by the data processor
* Formatted Data (or `Fit Ready Data` ?): data input to the fitter.

Reply:
Good suggestion. Thanks!